File: ent/schema/theme.go
package schema

import (
	"entgo.io/ent"
	"entgo.io/ent/schema/field"
)

type Theme struct {
	ent.Schema
}

func (Theme) Mixin() []ent.Mixin {
	return []ent.Mixin{
		BaseMixin{},
	}
}

func (Theme) Fields() []ent.Field {
	return []ent.Field{
		field.String("name").NotEmpty().Unique().MaxLen(100).Comment("主题名称，唯一标识"),
		field.String("display_name").NotEmpty().MaxLen(100).Comment("显示名称"),
		field.String("description").MaxLen(500).Optional().Comment("主题描述"),
		field.String("author_name").MaxLen(100).Optional().Comment("作者姓名"),
		field.String("author_email").MaxLen(100).Optional().Comment("作者邮箱"),
		field.String("logo").MaxLen(255).Optional().Comment("主题logo"),
		field.String("homepage").MaxLen(255).Optional().Comment("主页"),
		field.String("repo").MaxLen(255).Optional().Comment("仓库地址"),
		field.String("issue").MaxLen(255).Optional().Comment("问题反馈地址"),
		field.String("setting_name").MaxLen(100).Optional().Comment("设置名称"),
		field.String("config_map_name").MaxLen(100).Optional().Comment("配置映射名称"),
		field.String("version").NotEmpty().MaxLen(20).Comment("版本号"),
		field.String("require").MaxLen(50).Default("*").Comment("要求的gobee版本"),
		field.String("license").MaxLen(50).Optional().Comment("许可证"),
		field.String("path").NotEmpty().MaxLen(255).Comment("主题文件路径"),
		field.Bool("enabled").Default(false).Comment("是否启用"),
	}
}

func (Theme) Edges() []ent.Edge {
	return nil
}

File: ui/src/api/infra/theme.ts
import {http} from '@/utils/http'
import { BASE_URL, type ApiResponse, type TableResponse } from '@/api/utils'

export interface Theme {
  id: number
  created_at: string
  updated_at: string
  name: string
  display_name: string
  description?: string
  author_name?: string
  author_email?: string
  logo?: string
  homepage?: string
  repo?: string
  issue?: string
  setting_name?: string
  config_map_name?: string
  version: string
  require?: string
  license?: string
  path: string
  enabled: boolean
}

export interface ThemePageParams {
  page: number
  size: number
}

export const getThemePage = (params: ThemePageParams) => {
  return http.request<TableResponse<Theme>>('get', `${BASE_URL}/v1/theme/page`, { params })
}

export const queryTheme = (id: number) => {
  return http.request<ApiResponse<Theme>>('get', `${BASE_URL}/v1/theme/query/${id}`)
}

export const uploadTheme = (file: FormData) => {
  return http.request<ApiResponse<Theme>>('post', `${BASE_URL}/v1/theme/create`, { 
    data: file,
    headers: {
      'Content-Type': 'multipart/form-data'
    }
  })
}

export const deleteTheme = (id: number) => {
  return http.request<ApiResponse<null>>('delete', `${BASE_URL}/v1/theme/delete/${id}`)
}

export const enableTheme = (id: number) => {
  return http.request<ApiResponse<Theme>>('post', `${BASE_URL}/v1/theme/${id}/enable`)
}

export const disableTheme = (id: number) => {
  return http.request<ApiResponse<Theme>>('post', `${BASE_URL}/v1/theme/${id}/disable`)
}
